file/file_prefetch_buffer.cc

```diff
@@ -100,6 +108,8 @@ Status FilePrefetchBuffer::Read(const IOOptions& opts,
   }
 
   // Update the buffer offset and size.
+  // TODO: this is confusing for the direct I/O case as these should already
+  // be set properly
   bufs_[index].offset_ = rounddown_start;
```

Comment:
Can you elaborate on this TODO? We set the offset here after successfully reading the data.

Reply:
The line `buf->Size(buf->CurrentSize() + tmp.size());` in `RandomAccessFileReader::Read()` in its `use_direct_io()` branch

Reply:
OK I kind of gave up for now on having offset_ set prior to the Read(). I did ensure cursize_ is properly set prior to read though.